### PR TITLE
Remove deprecated behavior from App::path().

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Core/App.php [hunk 1]
```diff
@@ -187,7 +187,7 @@ protected static function _classExistsInBase(string $name, string $namespace): b
      */
     public static function path(string $type, ?string $plugin = null): array
     {
-        if ($plugin === null && $type[0] === strtolower($type[0])) {
+        if ($plugin === null) {
             return (array)Configure::read('App.paths.' . $type);
```
Comment (garas) [inline src/Core/App.php:191]: Both `/** @psalm-suppress PossiblyNullArgument */` can now be removed below because `$plugin` never `null`.

Comment (markstory) [reply src/Core/App.php:191]: Couldn't a null still be provided as an argument?

Comment (garas) [reply src/Core/App.php:191]: When `$plugin === null`, function exits here, so few lines below (GitHub didn't let comment on them directly) `Plugin::templatePath($plugin)` and `Plugin::path($plugin)` always passes `string`.

Comment (ADmad) [reply src/Core/App.php:191]: Yup I'll remove those annotations.

**Pull request merged**